confidant/routes/v1.py

```diff
@@ -377,21 +392,27 @@ def get_credential(id):
     if (cred.data_type != 'credential' and
             cred.data_type != 'archive-credential'):
         return jsonify({}), 404
+
     services = []
     for service in Service.data_type_date_index.query('service'):
         services.append(service.id)
     if cred.data_type == 'credential':
         context = id
     else:
         context = id.split('-')[0]
-    data_key = keymanager.decrypt_datakey(
-        cred.data_key,
-        encryption_context={'id': context}
-    )
-    cipher_version = cred.cipher_version
-    cipher = CipherManager(data_key, cipher_version)
-    _credential_pairs = cipher.decrypt(cred.credential_pairs)
-    _credential_pairs = json.loads(_credential_pairs)
+
+    _credential_pairs = {}
+    _is_authorized = False
+    if authnz.user_is_authorized(cred.group):
+        _is_authorized = True
+        data_key = keymanager.decrypt_datakey(
+            cred.data_key,
+            encryption_context={'id': context}
+        )
+        cipher_version = cred.cipher_version
+        cipher = CipherManager(data_key, cipher_version)
+        _credential_pairs = cipher.decrypt(cred.credential_pairs)
+        _credential_pairs = json.loads(_credential_pairs)
```

Comment:
The newer version of the code handles this logic currently. It will only deliver decrypted credentials if the correct ACL permissions exist.